sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedColumnReader.java

```diff
@@ -146,10 +153,8 @@ void readBatch(int total, ColumnVector column) throws IOException {
       int num = Math.min(total, leftInPage);
       if (useDictionary) {
         // Read and decode dictionary ids.
-        ColumnVector dictionaryIds = column.reserveDictionaryIds(total);
         defColumn.readIntegers(
             num, dictionaryIds, column, rowId, maxDefLevel, (VectorizedValuesReader) dataColumn);
-
         if (column.hasDictionary() || (rowId == 0 &&
```

Comment:
In the original fix, do you know the reasoning behind calling `column.setDictionary()` when the column already has an existing dictionary? That seems incorrect to me.